### ISSUE TITLE
Messages with inline attachments aren't found with flag:a

### ISSUE THREAD

**Issue by dkogan** [djcb/mu#927]
Just what the subject says: 'mu find g:a' finds messages with binary attachments, but not inline ones. Is this a known bug?

Thanks!

**Comment by djcb** [comment]
It's not a bug because inline mime-parts are by definition not attachments ;-). But actually mu considers some inline parts as attachments as well. Anyway, can't do much unless you provide an example raw message where you think the 'attachment' is not correctly detected -- thanks!

**Comment by dkogan** [comment]
Dirk-Jan C. Binnema <notifications@github.com> writes:

> It's not a bug because inline mime-parts are by definition not
> attachments ;-). But actually mu considers some inline parts as
> attachments as well. Anyway, can't do much unless you provide an
> example raw message where you think the 'attachment' is not correctly
> detected -- thanks!

Hi. I think g:a should detect inline "attachments" as well because:

1. A user sending email with mu4e:compose mode generates both types of
attachments in the same way: C-c RET f

2. When mu4e displays a message, both types of attachments are displayed
in the same way

I'm including a simple inline attachment to this message.

Thanks!

**Comment by djcb** [comment]
As I said, please attach a *raw message* where you think mu should recognize an attachment, but it doesn't.

**Comment by dkogan** [comment]
Dirk-Jan C. Binnema <notifications@github.com> writes:

> As I said, please attach a raw message where you think mu should
> recognize an attachment, but it doesn't.

Huh. The little attachment I sent earlier in this thread WAS detected
correctly. I'm attaching a raw message from the emacs mailing list where
a patch was inline-attached, and mu4e didn't set the "attachment" flag

**Comment by djcb** [comment]
Hmm, can't find the attachment... where did you put it? It should be attached to this issue report (e.g. using drag & drop).

**Comment by dkogan** [comment]
Dirk-Jan C. Binnema <notifications@github.com> writes:

> Hmm, can't find the attachment... where did you put it? It should be attached
> to this issue report (e.g. using drag & drop).

Drag&drop??? I'm using mu4e and emacs! :)

Apparently github is retarded, and silently throws out attachments.
Doing it via their clunky ui....

**Comment by dkogan** [comment]
Never mind. Their this is REALLY retarded. Have a link:

http://pub.secretsauce.net/raw_message_inline_attachment_undetected.msg

**Comment by djcb** [comment]
Thanks. I've made a change so that patches/diffs are shown as part of the body.

**Comment by dkogan** [comment]
Dirk-Jan C. Binnema <notifications@github.com> writes:

> Thanks. I've made a change so that patches/diffs are shown as part of
> the body.

Hi. I just got a chance to play with this change. It works as
advertised, but I don't think it solves the fundamental problem. The
problem is that

- To the user, inline attachments look no different than any other
  attachment:

  - inline "attachments" are created the same way as any other
  attachment: (mml-attach-file)

  - mu4e UI displays them as any other attachment: there's an
  'Attachments:' field in the *mu4e-view* buffer

- mu4e does NOT set the 'attachment' flag on messages containing these

- such messages are not findable with the 'flag:a' search


I know I'm not giving you a patch, so this is a bit of an empty
complaint, but it's a valid one, at least (I think).

Thanks
dima

**Comment by dkogan** [comment]
Hi. So I just sat down to make a patch, and it became clear to me that this functionality (inline attachments setting the "attachment" flag) already exists. The reason this wasn't apparent is that I had indexed my mail prior to when this was implemented, and subsequent invocations of "mu index" were NOT adding this flag. Removing the database in ~/.mu and reindexing everything from scratch made the flags correct. No idea when this was fixed, but thanks a lot for doing that!
